src/Microsoft.IdentityModel.Logging/LogHelper.cs

```diff
@@ -37,6 +39,36 @@ namespace Microsoft.IdentityModel.Logging
     /// </summary>
     public class LogHelper
     {
+        /// <summary>
+        /// Gets or sets a logger to which logs will be written to.
+        /// </summary>
+        public static IIdentityLogger Logger { get; set; } = NullIdentityModelLogger.Instance;
+
+        /// <summary>
+        /// Indicates whether or the log message header (contains library version, date/time, and PII debugging information) has been written.
+        /// </summary>
+        private static bool HeaderWritten { get; set; } = false;
+
+        /// <summary>
+        /// The log message that indicates the current library version.
+        /// </summary>
+        private static string _versionLogMessage = "Library version of Microsoft.IdentityModel: {0}.";
+
+        /// <summary>
+        /// The log message that indicates the date.
+        /// </summary>
+        private static string _dateLogMessage = "Date: {0}.";
+
+        /// <summary>
+        /// The log message that is shown when PII is off.
+        /// </summary>
+        private static string _piiOffLogMessage = "PII (personally identifiable information) logging is currently turned off. Set IdentityModelEventSource.ShowPII to 'true' to view the full details of exceptions.";
```

Comment:
Logs can become very noisy fast...You might just have:
"PII (personally identifiable information) logging is currently: {0}. See [https://aka.ms/id-model-pii](https://aka.ms/id-model-pii). " and pass in a bool

I just made up that aka.ms link, but there you could have more details.

Reply:
Good idea, I'll look into how to create a aka.ms link and shorten the message.